pkg/registration/spoke/spokeagent.go

```diff
@@ -54,9 +54,8 @@ type SpokeAgentConfig struct {
 
 	internalHubConfigValidFunc wait.ConditionWithContextFunc
 
-	hubKubeConfigChecker             *hubKubeConfigHealthChecker
-	bootstrapKubeconfigHealthChecker *bootstrapKubeconfigHealthChecker
-	reSelectChecker                  *reSelectChecker
+	hubKubeConfigChecker *hubKubeConfigHealthChecker
+	reSelectChecker      *reSelectChecker
```

Comment:
@xuezhaojun

Just thought in the `multiplehubs` case, we also need an immediate restart as well. For example, when hub1 is down for 3 min (the threshold we set), we expect a restart and reselection to happen at that time point, not a few minutes later based on the health checker.

Maybe we should refactor these checkers all into using this `cancel` approach(but not in this PR, a follow-up PR in the future)? @qiujian16 any thoughts on this?

Reply:
yes, we could create an issue for followup